src/acl.c

```diff
@@ -976,22 +974,10 @@ int ACLSetUser(user *u, const char *op, ssize_t oplen) {
                 return C_ERR;
             }
 
-            /* The command should not be set right now in the command
-             * bitmap, because adding a subcommand of a fully added
-             * command is probably an error on the user side. */
             unsigned long id = ACLGetCommandID(copy);
-            if (ACLGetUserCommandBit(u,id) == 1) {
```

Comment:
You're still adding the subcommand to the list below, I think we need to return here if we detect the bit is already set.

Reply:
Yes unnecessary memory allocation is happening.